### ISSUE TITLE
validation problem, xmldsig# namespace for XML digital signature

### ISSUE THREAD

**Issue by brutzman** [sissaschool/xmlschema#357]
xmlschema is flagging the namespace for XML digital signature since it ends with a hash tag #.

*  `xmlns="http://www.w3.org/2000/09/xmldsig#"`

or
*  `xmlns:ds="http://www.w3.org/2000/09/xmldsig#"`

The construct itself is OK, and used in XML documents with a digital signature, as specified by authoritative reference

* XML Signature Syntax and Processing Version 2.0, W3C Working Group Note 23 July 2015
* http://www.w3.org/TR/xmldsig-core2
* https://www.w3.org/TR/xmldsig-core2/#def-XMLnamespaceURI

This namespace is used by package x3d.py as part of full round-trip support between XML and Python for X3D models.
* https://pypi.org/project/x3d

For repeatability, example error:
* https://www.web3d.org/x3d/content/examples/Basic/ChemicalMarkupLanguage/CaffeineIndex.html
* https://www.web3d.org/x3d/content/examples/Basic/ChemicalMarkupLanguage/Caffeine.py

```
$ python Caffeine.py
x3d.py package 4.0.64.4 loaded, have fun with X3D Graphics!
Self-test diagnostics:
*** Python-to-XML well-formed XML document test failed.
    x3dSchemaUrl=https://www.web3d.org/specifications/x3d-4.0.xsd
    XMLSchemaParseError cannot import namespace 'http://www.w3.org/2000/09/xmldsig#': Entities are forbidden (entity_name='dsig')
```

Thanks for considering acceptance of this important namespace URI.

**Comment by brunato** [comment]
Hi,

this would be related with the [*defuse* option](https://xmlschema.readthedocs.io/en/latest/features.html#xml-entity-based-attacks-protection). 

For default remote content is parsed with a safe XML parser that reject XML data containing entity declarations.

Infact the XSD schema of the "http://www.w3.org/2000/09/xmldsig#" namespace, imported from  https://www.w3.org/TR/2002/REC-xmldsig-core-20020212/xmldsig-core-schema.xsd contains an entity declaration, so the safe parser raises an error.

As a workaround you can provide `defuse='never'` when the schema instance is created into *Caffeine.py* module. I'll evaluate the inclusion of "http://www.w3.org/2000/09/xmldsig#" namespace either into fallback locations or as a special case.

**Comment by brunato** [comment]
Hi,
in the release v2.4.0 the schemas for XML signature and encryption (dsig/dsig11, xenc/xenc11) have been added to local fallback schemas.

Also, if the safe XML parser rejects the loading of a schema (like in your case) this is not handled as a _not parsable_ error (as in the previous version) but as a _location is not accessible_ error, so the import fallback location is processed (if any).
The local copy of dsig/xenc schemas do not include entity declarations in order to avoid defusing in any case.

Please check if your package works with the version v2.4.0, and in this case close the issue.

Thank you

**Comment by brutzman** [comment]
Apologies to miss this earlier.  Have tested with xmlschema-2.4.0 and the problem no longer occurs.

Looks great, thanks!
